### PR TITLE
Move documentation to reStructuredText format

### PR DESCRIPTION
**Pull request by greg0ire**
## Subject

The README is going to be wiped out by dev-kit, and it might be nice to
display this on the website.

### CODE REVIEW

File: Resources/doc/index.rst [hunk 1]
```diff
@@ -1,17 +1,30 @@
 Sonata Admin Search Bundle
 ==========================
 
+By default, SonataAdminBundle_ uses the storage backend full-text search
+capabilities to provide search results or filtered listings. This bundle should
+help you leverage the power of your search engine when full-text search is not
+good enough.
+
+For the moment, only elasticsearch_ is supported, with the help of
+FOSElasticaBundle_ finder services.
```
Comment (Soullivaneuh) [inline Resources/doc/index.rst]: Missing new line after this.

File: Resources/doc/reference/configuration.rst [hunk 2]
```diff
@@ -0,0 +1,12 @@
+Configuration
+-------------
+
+You need to map each admin to a FOS finder service that should be used for that
+admin.
+
+.. code-block:: yaml
+    sonata_admin_search:
```
Comment (soullivaneuh) [inline Resources/doc/reference/configuration.rst:9]: Should not have a blank line between the `code-block` and the content?

File: README.md [hunk 3]
```diff
@@ -2,37 +2,7 @@
...
-        my_admin.id: # Admin service id
-            finder: Id for a FOS finder service that should be used # Finder service
-            actions : [list] #[Optional] actions where elasticsearch has to be enabled
-```
 
 ## Documentation
 
 For contribution to the documentation you cand find it on [Resources/doc](https://github.com/sonata-project/SonataAdminSearchBundle/tree/master/Resources/doc).
-
-[0]:https://travis-ci.org/sonata-project/SonataAdminSearchBundle.svg?branch=master
```
Comment (soullivaneuh) [inline README.md:34]: This link is still used.

File: README.md [hunk 4]
```diff
@@ -2,37 +2,7 @@
...
-[1]:https://travis-ci.org/sonata-project/SonataAdminSearchBundle
```
Comment (soullivaneuh) [inline README.md:35]: This link is still used.

File: Resources/doc/reference/configuration.rst (hunk 2 above)
Comment (greg0ire) [inline Resources/doc/reference/configuration.rst:9]: I don't know, so let's do it.

File: README.md [hunk 5]
```diff
@@ -2,37 +2,7 @@
 
 [![Build Status][0]][1]
 
-By default, [SonataAdminBundle][2] uses the storage backend full-text search
```
Comment (soullivaneuh) [inline README.md:5]: Better idea: Don't touch the README right now. Dev-kit will wipe this out.

Comment (greg0ire) [inline README.md:5]: Nah, I'm done anyway, so… plus, no need to have duplicate data.

**Comment by soullivaneuh** [comment]
Thank you! :+1:

**Pull request merged**